electrum/plugin.py

```diff
@@ -543,6 +552,8 @@ def client_by_xpub(self, plugin: 'HW_PluginBase', xpub, handler: 'HardwareHandle
                        devices: Sequence['Device']) -> Optional['HardwareClientBase']:
         _id = self.xpub_id(xpub)
         client = self._client_by_id(_id)
+        if client and not self._type_check_client_plugin(client, plugin):
```

Comment:
probably cleaner to move this check one line below, to be at the top of the existing `if client` block

Reply:
Ah sorry, I meant like this:
```python
        if client:
            if not self._type_check_client_plugin(client, plugin):
                return
            # An unpaired client might have another wallet's handler
            # from a prior scan.  Replace to fix dialog parenting.
            client.handler = handler
            return client
```
(but see other comment about the helper function)